### PR TITLE
Fix instance where evolve_all is unicode - fixes #2281

### LINKED ISSUE #2281: evolve_all "all" is not working
### Expected Behavior
Bot would try to evolve all eligible pokemon when "evolve_all": "all" is set

### Actual Behavior
Bot does not evolve anything when "evolve_all": "all" is set.  If I set it to "evolve_all": "all,Weedle" I get the following output: 

```
[19:11:08]  Starting PokemonGo Bot....
[19:11:09]  [x] Current egg hatches in 0.85 km
[19:11:10] [#] Successfully evolved Weedle with 180 CP and 0.4 IV!
[19:11:13] [#] Successfully evolved Weedle with 113 CP and 0.51 IV!
```

### Steps to Reproduce
Set "evolve_all" to all on this commit

### Other Information
OS: Ubuntu 14.04.4 LTS
Git Commit: eee7ba49c5c4e6bb1b3efefcf9f99a7c72a48671
Python Version: python 2.7.6

### PR DESCRIPTION
**Pull request by JSchwerberg**
Short Description: 

allows evolve_all to be Unicode, and runs the in-built _validate_config function which is currently not being called

Fixes:
- #2281 
- 
-

### CODE REVIEW

**Comment by mrthrowaway** [comment]
Tested. This works for me

**Comment by mrthrowaway** [comment]
Looks like the same changes are also made in the refactor of evolve_all in pull #2244

**Comment by JSchwerberg** [comment]
Yep, I see those as well; as well as an internal check to see if you can evolve pokemon rather than slamming the API.  #2244 is unresolved, though, and waiting on bugfixes -- will wait for dev to let me know how to proceed.

**Comment by riseno** [comment]
Work perfectly for me. nice work.

**Comment by pokecunt** [comment]
download

**Comment by inzaghi89** [comment]
Awesome, works as a champ :+1:

**Comment by ewold** [comment]
Tested works for me 👍

**Comment by ewold** [comment]
Although after the evolve bot wouldn't find pokemons anymore, don't know if it was just me , but after turning evolve_all to NONE the bot instantly found them.

**Comment by inzaghi89** [comment]
No issue for me.

File: pokemongo_bot/cell_workers/evolve_all.py [hunk 1]
```diff
@@ -9,9 +9,10 @@ def initialize(self):
         self.evolve_speed = self.config.get('evolve_speed', 3.7)
         self.evolve_cp_min = self.config.get('evolve_cp_min', 300)
         self.use_lucky_egg = self.config.get('use_lucky_egg', False)
+        self._validate_config()
 
     def _validate_config(self):
-        if isinstance(self.evolve_all, str):
+        if isinstance(self.evolve_all, str) or isinstance(self.evolve_all, unicode):
```
Comment (douglascamata) [inline pokemongo_bot/cell_workers/evolve_all.py]: Check if both are an instance of `basestring` instead. `str` and `unicode` inherits from `basestring`.

**Comment by JSchwerberg** [comment]
@douglascamata New commit that fixes what you pointed out; thanks for that, didn't think of basestring!

**Pull request merged**